HB/common/utils.elpi

```diff
@@ -93,17 +93,17 @@ gref->modname GR ModName :-
 % function to predicate generic constructions %
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
-pred mk-nil o:any.
+pred mk-nil o:list A.
 mk-nil [].
-pred mk0 i:any, o:any.
+pred mk0 i:A, o:A.
 mk0 F R :- constant R F [].
-pred mk1 i:any, i:any, o:any.
+pred mk1 i:(A -> B), i:A, o:B.
 mk1 F X1 R :- constant R F [X1].
-pred mk2 i:any, i:any, i:any, o:any.
+pred mk2 i:(A -> B -> C), i:A, i:B, o:C.
 mk2 F X1 X2 R :- constant R F [X1, X2].
-pred mk3 i:any, i:any, i:any, i:any, o:any.
+pred mk3 i:(A -> B -> C -> D), i:A, i:B, i:C, o:D.
 mk3 F X1 X2 X3 R :- constant R F [X1, X2, X3].
-pred mk4 i:any, i:any, i:any, i:any, i:any, o:any.
+pred mk4 i:(A -> B -> C -> D -> E), i:A, i:B, i:C, i:D, o:E.
```

Comment:
This loose typing made me lose a crapload of time, now this is more strict and indeed it identified the problem in `w-params.map` which is fixed "the verbose way".

Reply:
ok, great